### PR TITLE
mkuplus and mkuminus must preserve attributes

### PR DESCRIPTION
**Pull request by goldfirere**
Review: @ncik-roberts

### CODE REVIEW

File: ocaml/parsing/parser.mly [hunk 1]
```diff
@@ -127,7 +127,8 @@ let neg_string f =
   then String.sub f 1 (String.length f - 1)
   else "-" ^ f
 
-let mkuminus ~oploc name arg =
+let mkuminus ~loc ~oploc name arg =
```
Comment (ncik-roberts) [inline ocaml/parsing/parser.mly:130]: I believe this duplicates the attributes in the non-constant case. (Same for `mkuplus`.)

Comment (goldfirere) [reply ocaml/parsing/parser.mly:130]: Why do you believe that? I don't see it. This function is essentially decomposing the input `arg` and using only its `desc`. My change also preserves its attributes.

Comment (ncik-roberts) [reply ocaml/parsing/parser.mly:130]: The full implementation is:

```ocaml
let mkuminus ~loc ~oploc name arg =
  mkexp ~loc ~attrs:arg.pexp_attributes @@
  match name, arg.pexp_desc with
  | "-", Pexp_constant(Pconst_integer (n,m)) ->
      Pexp_constant(Pconst_integer(neg_string n,m))
  | ("-" | "-."), Pexp_constant(Pconst_float (f, m)) ->
      Pexp_constant(Pconst_float(neg_string f, m))
  | _ ->
      Pexp_apply(mkoperator ~loc:oploc ("~" ^ name), [Nolabel, arg])
```

Note how the `mkexp ~loc ~attrs:arg.pexp_attributes` can be applied to a `Pexp_apply` node that has `arg` as a subtree.

Comment (goldfirere) [reply ocaml/parsing/parser.mly:130]: Oh -- I missed that. You're right. Will fix shortly.

**Review by ncik-roberts — APPROVED** [review]
Thanks for fixing this.

**Pull request merged**